jetty-alpn/jetty-alpn-client/src/main/java/org/eclipse/jetty/alpn/client/ALPNClientConnectionFactory.java

```diff
@@ -54,21 +54,8 @@ public ALPNClientConnectionFactory(Executor executor, ClientConnectionFactory co
         IllegalStateException failure = new IllegalStateException("No Client ALPNProcessors!");
 
         // Use a for loop on iterator so load exceptions can be caught and ignored
-        for (Iterator<Client> i = ServiceLoader.load(Client.class).iterator(); i.hasNext(); )
+        TypeUtil.serviceLoaderStream(ServiceLoader.load(Client.class)).flatMap(TypeUtil::providerMap).forEach((processor) ->
```

Comment:
Perhaps it would be less verbose to have two methods named:
 - `Stream<T> TypeUtil.serviceStream(ServiceLoader<T>)` would include in the flatMap operation and just provide a stream of services, hiding exceptions
 - `Stream<Provider<T>> TypeUtil.serviceProviderStream(ServiceLoader<T>)` would return the stream of providers with exceptions etc.

Reply:
What's the point of the second one?

That's what `ServiceLoader.stream()` already returns.

https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/ServiceLoader.html#stream()

Reply:
@joakime our version actually works!   The ServiceLoader stream implementation has a bug, because it uses the iterator internally and does not protect against the cases that `hasNext` can throw.    If the JVM folks accept this is a bug and undertake to fix it, then we can deprecate the second option and eventually replace it.